[iOS] Label LinebreakMode (TailTruncation) for FormattedText does't work in CollectionView after scroll - fix

kubaflo opened the pull request:
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/28147

|Before|After|
|--|--|
|<video src="https://github.com/user-attachments/assets/09a75432-4b03-465a-910d-844076f18548" width="300px"/>|<video src="https://github.com/user-attachments/assets/29d23e8c-6eaf-45c0-a6ed-e90ffc089c52" width="300px"/>|

jsuarezruiz commented:
/azp run

jsuarezruiz commented on src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue28147.cs:
```diff
@@ -0,0 +1,23 @@
+	{
+		App.WaitForElement("CollectionView");
+		App.ScrollDown("CollectionView", ScrollStrategy.Gesture, 0.80);
+		VerifyScreenshot();
```
Pending snapshots already available in the latest build. Example, iOS:
![image](https://github.com/user-attachments/assets/c07e5515-084e-4897-84b5-ed28ef5e4728)

Could you commit the images?

kubaflo replied:
Commited

jsuarezruiz commented:
/azp run

jsuarezruiz reviewed (changes requested):
The test **LineBreakModeShouldWorkInCollectionViewAfterScrolling** is failing on Android, iOS and Windows. Differences in the Label position between the images.

Example:
![image](https://github.com/user-attachments/assets/d6f81025-0d7f-4724-a119-3f8ce9e9f275)

Could you review it?

kubaflo commented:
/azp run maui-pr-uitests

kubaflo committed:
Propagate LineBreakMode to iOS attributed strings

Add LineBreakMode support to FormattedString/Span NSAttributedString conversions on iOS. New internal overloads accept a LineBreakMode (defaulting to WordWrap) and the existing callers were updated to pass label.LineBreakMode. The Span conversion maps Forms LineBreakMode to the corresponding UILineBreakMode so attributed strings respect the requested wrapping/truncation behavior.
